BSP/STM32F413/Inc/BSP_CAN.h

```diff
@@ -0,0 +1,35 @@
+#ifndef __BSP_CAN_H__
+#define __BSP_CAN_H__
+
+#include <stdbool.h>
+#include "common.h"
+#include "stm32f4xx_hal.h"
+#include "FreeRTOS.h"
+#include "queue.h"
+#include "semphr.h"
+#include "CANMetaData.h"
+
+typedef struct CAN_TxPayload
+{
+    CAN_TxHeaderTypeDef header; /* can transmit header info */
+    uint32_t mailbox;           /* has which mailbox used when sent */
+    uint8_t data[8];            /* can data */
+} CAN_TxPayload_t;
+
+typedef struct CAN_RxPayload
+{
+    CAN_RxHeaderTypeDef header; /* can recieve header info */
+    uint8_t data[8];            /* can data */
+} CAN_RxPayload_t;
+
+/* handlers */
+extern const CAN_HandleTypeDef *CarCAN;
+extern const CAN_HandleTypeDef *LocalCAN;
+
+void CAN_CarCANInit(void);
+void CAN_LocalCANInit(void);
+bool CAN_Send(const CAN_HandleTypeDef *can, CAN_TxPayload_t payload, bool blocking);
+bool CAN_SetListener(const CAN_HandleTypeDef *can, CANID_t id, QueueHandle_t *queue);
+void CAN_ClearListener(const CAN_HandleTypeDef *can, CANID_t id);
```

Comment:
Can we add comments indicating what each of these do? params for set/clear are a little unclear